src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -54,137 +117,881 @@ static ColliderCylinderInit D_80B90E54 = {
+        } else {
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+    } else if ((Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) || (mREG(1) == 1)) {
+        ObjBean_SetupWaitForWater(this);
+    } else {
+        ObjBean_SetupWaitForBean(this); // Soft soil spot
+    }
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.home.rot.z = this->dyna.actor.shape.rot.z;
+    this->dyna.actor.world.rot.z = this->dyna.actor.shape.rot.z;
+    // Magic bean tree lift
+    osSyncPrintf("(魔法の豆の木リフト)(arg_data 0x%04x)\n", this->dyna.actor.params);
+}
+
+void ObjBean_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBean* this = THIS;
+
+    if (this->stateFlags & DYNAPOLY_SET) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+    if (this->stateFlags & COLLIDER_SET) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+    if (this == D_80B90E30) {
+        D_80B90E30 = NULL;
+    }
+}
+
+void ObjBean_SetupWaitForBean(ObjBean* this) {
+    this->actionFunc = ObjBean_WaitForBean;
+    ObjBean_SetDrawMode(this, DRAW_LEAVES);
+    this->dyna.actor.textId = 0x2F;
+}
+
+void ObjBean_WaitForBean(ObjBean* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->dyna.actor, globalCtx)) {
+        if (func_8002F368(globalCtx) == EXCH_ITEM_BEAN) {
+            ObjBean_SetupPlantPhase1(this);
+            Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+        }
+    } else {
+        func_8002F298(&this->dyna.actor, globalCtx, 40.0f, EXCH_ITEM_BEAN);
+    }
+}
+
+void ObjBean_SetupPlantPhase1(ObjBean* this) {
+    this->actionFunc = ObjBean_PlantPhase1;
+    ObjBean_SetDrawMode(this, DRAW_LEAVES);
+    this->timer = 60;
+}
+
+// Link is looking at the soft soil
+void ObjBean_PlantPhase1(ObjBean* this, GlobalContext* globalCtx) {
+    if (this->timer <= 0) {
+        ObjBean_SetupPlantPhase2(this);
+    }
+}
+
+void ObjBean_SetupPlantPhase2(ObjBean* this) {
+    this->actionFunc = ObjBean_PlantPhase2;
+    ObjBean_SetDrawMode(this, DRAW_LEAVES | DRAW_SOIL);
+    Actor_SetScale(&this->dyna.actor, 0.01f);
+}
+
+// The leaves are visable and growing
+void ObjBean_PlantPhase2(ObjBean* this, GlobalContext* globalCtx) {
+    s32 temp_v1;
+
+    temp_v1 = Math_StepToF(&this->dyna.actor.scale.y, 0.16672663f, 0.01f) & 1;
+    temp_v1 &= Math_StepToF(&this->dyna.actor.scale.x, 0.03569199f, 0.00113f);
+
+    this->dyna.actor.scale.z = this->dyna.actor.scale.x;
+    if (temp_v1) {
+        if (this->timer <= 0) {
+            func_80B8FF50(this);
+        }
+    } else {
+        this->timer = 1;
+    }
+    func_8002F974(&this->dyna.actor, NA_SE_PL_PLANT_GROW_UP - SFX_FLAG);
+}
+
+void func_80B8FF50(ObjBean* this) {
+    this->actionFunc = func_80B8FF8C;
+    ObjBean_SetDrawMode(this, DRAW_LEAVES | DRAW_SOIL);
+    this->unk_1B6.x = 0x33E9;
+}
+
+void func_80B8FF8C(ObjBean* this, GlobalContext* globalCtx) {
```

Comment:
For this I was thinking since I had missed it I was going to wait until I came up with better names for these since I dont like the names they have right now.

Reply:
Yes, but I'd rather the interim names are at least consistent, until we come up with better ones: as they are now, the original address-based names arguably provide more useful information (e.g. if someone's going to breakpoint them to analyse the behaviour).